src/archives.rs

```diff
@@ -13,9 +13,11 @@ pub(crate) enum ArchiveFormat {
 impl ArchiveFormat {
     /// Parse archive type from resource extension.
     pub(crate) fn parse_from_extension(resource: &str) -> Result<Self, Error> {
```

Comment:
Since we're inferring the file type based on content instead of the extension, we should probably rename this method. Maybe something like 
```suggestion
    pub(crate) fn parse_from_path<P: AsRef<Path>>(path: P) -> Result<Self, Error> {
```